### ISSUE TITLE
Follow python-dotenv syntax to resolve process environment variables

### ISSUE THREAD

**Issue by karrtikr** [microsoft/debugpy#821]
In reference to https://github.com/microsoft/vscode-python/issues/18307 & https://github.com/microsoft/debugpy/issues/820#issuecomment-655799640. Wrapping the environment variable in `${env:<VARNAME>}` is currently required to resolve env variables in process's environment. With this we add support to resolve it using `${<VARNAME>}`.

Another solution could be to use environment variable substitution/resolution support in the extension where we already follow this syntax. We can expose [environment variable fetcher](https://github.com/microsoft/vscode-python/blob/456ac068e95ba600079ddcdc0cc7551d1659a699/src/client/common/variables/environmentVariablesProvider.ts#L56) as an API and have the debugger use it to fetch env variables.

**Comment by fabioz** [comment]
@karrtikr @int19h right now I think that the debugger doesn't really do any variable substitution (the environment is expected to be passed from the extension).

Is the idea starting to do that?

**Comment by karrtikr** [comment]
If it's possible yes that would be ideal.

**Comment by karrtikr** [comment]
Is it possible to pass env variables through the extension? If yes that would be ideal.

**Comment by karthiknadig** [comment]
yes, you can pass environment variables to the debugger. We can do this in debug configuration resolver.

**Comment by karthiknadig** [comment]
I don't think we should do this work in the debugger at all. This can be entirely done on the extension side.

**Comment by karrtikr** [comment]
Cool, can you transfer this back?
